tools/GulpUtils.ts

```diff
@@ -15,13 +15,29 @@ import Q = require ("q");
 
 class GulpUtils {
     private static testCommand: string = "test";
+    private static testDir: string = "test"; // test directory
+    private static mochaOptFile: string = "mocha.opts"; // mocha config file
+    private static testCordovaExemptText: string = "TestCordovaExempt"; // test directory
+    private static mochaTestInvert: string = "--invert --grep"; // inverts --grep matches
 
-    public static runAllTests(modulesToTest: string[], modulesRoot: string, failTestsAtEnd: boolean, reporter: string): Q.Promise<any> {
+    public static runAllTests(modulesToTest: string[], modulesRoot: string, failTestsAtEnd: boolean, reporter: string, testCordova: string): Q.Promise<any> {
         var args: string[] = [];
         if (reporter) {
             args = ["--reporter", reporter];
         }
-      return GulpUtils.runNpmScript(modulesToTest, modulesRoot, GulpUtils.testCommand, failTestsAtEnd, args);
+
+        return GulpUtils.chainAsync<string>(modulesToTest, (moduleName: string) => {
+
+            if (testCordova) {
+                var mochaOptFile: string = path.resolve(modulesRoot, moduleName, GulpUtils.testDir, GulpUtils.mochaOptFile);
+                fs.appendFileSync(mochaOptFile, " " + GulpUtils.mochaTestInvert + " " + GulpUtils.testCordovaExemptText); // i.e. run all tests except for TestCordovaExempt
+                process.env.testCordovaVersion = testCordova;
```

Comment:
Should this assignment to process.env.testCordovaVersion just happen once? And then we could `return Q.denodeify(fs.appendFile)(mochaOptFile, ...);` in the case that we modify things rather than always returning dummy promises.